ihs/src/main/scripts/was-check.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+#      Copyright (c) Microsoft Corporation.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+# 
+#           http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Get tWAS installation properties
+source /datadrive/virtualimage.properties
+
+echo "Checking at + $(date)" > /var/log/cloud-init-was.log
+
+# Read custom data from ovf-env.xml
+customData=`xmllint --xpath "//*[local-name()='Environment']/*[local-name()='ProvisioningSection']/*[local-name()='LinuxProvisioningConfigurationSet']/*[local-name()='CustomData']/text()" /var/lib/waagent/ovf-env.xml`
+IFS=',' read -r -a ibmIdCredentials <<< "$(echo $customData | base64 -d)"
+
+# Check whether IBMid is entitled or not
+entitled=false
+if [ ${#ibmIdCredentials[@]} -eq 2 ]; then
+    userName=${ibmIdCredentials[0]}
+    password=${ibmIdCredentials[1]}
+    
+    ${IM_INSTALL_DIRECTORY}/eclipse/tools/imutilsc saveCredential -secureStorageFile storage_file \
+        -userName "$userName" -userPassword "$password" -passportAdvantage
+    if [ $? -eq 0 ]; then
+        output=$(${IM_INSTALL_DIRECTORY}/eclipse/tools/imcl listAvailablePackages -cPA -secureStorageFile storage_file)
+        echo $output | grep -q "$WAS_ND_VERSION_ENTITLED" && entitled=true
+    else
+        echo "Cannot connect to Passport Advantage." >> /var/log/cloud-init-was.log
+    fi
+else
+    echo "Invalid input format." >> /var/log/cloud-init-was.log
+fi
+
+if [ ${entitled} = true ]; then
+    # Update all packages for the entitled user
+    output=$(${IM_INSTALL_DIRECTORY}/eclipse/tools/imcl updateAll -repositories "$REPOSITORY_URL" \
+        -acceptLicense -log log_file -installFixes none -secureStorageFile storage_file -preferences $SSL_PREF,$DOWNLOAD_PREF -showProgress)
```

Comment:
This should be `-installFixes recommended`, right?

Reply:
@git4rk  I have addressed your comments in commit 81bd04d.